### ISSUE TITLE
Slicing FileSequences is not supported

### ISSUE THREAD

**Issue by assumptionsoup** [sqlboy/fileseq#60]
```
>>> fs = fileseq.FileSequence('/my/path/image.#.jpg')

>>> fs.setFrameRange('1-12')

>>> fs[0]
'/my/path\\image.0001.jpg'

>>> fs[1]
'/my/path\\image.0002.jpg'

>>> fs[-1]
'/my/path\\image.0012.jpg'

>>> fs[len(fs)-2]
'/my/path\\image.0011.jpg'

>>> fs[1:]
Traceback (most recent call last):

  File "<ipython-input-27-8cd8dd826378>", line 1, in <module>
    fs[1:]

  File "C:\Users\jordan\AppData\Local\Continuum\anaconda2\lib\site-packages\fileseq\filesequence.py", line 362, in __getitem__
    return self.index(idx)

  File "C:\Users\jordan\AppData\Local\Continuum\anaconda2\lib\site-packages\fileseq\filesequence.py", line 336, in index
    return self.frame(self._frameSet[idx])

  File "C:\Users\jordan\AppData\Local\Continuum\anaconda2\lib\site-packages\fileseq\filesequence.py", line 313, in frame
    zframe = str(int(frame)).zfill(self._zfill)

TypeError: int() argument must be a string or a number, not 'tuple'
```

**Comment by justinfx** [comment]
I've got a patch for this in #61. I just want to confirm that you would expect the result type to be a list of strings, like standard python container behaviour. Or would you have expected it to have formatted a new sequence string using the range from the slice?

**Comment by assumptionsoup** [comment]
Thanks Justin. I would expect the output of slicing to be the same as indexing.

These would be equivalent:
```
fs = fileseq.FileSequence('/my/path/image.#.jpg')
for x in range(len(fs) // 2, len(fs)):
    print repr(fs[x])

for frame in fs[len(fs) // 2:]:
    print repr(frame)
```

**Comment by assumptionsoup** [comment]
Oh, wait. Were you asking about the type returned by the slice ` fs[len(fs) // 2:]`, rather than the elements?  I think I it would be nice to get a single new file sequence with the appropriate FrameSet, instead of a list of strings.  But that's just icing on the cake :)

**Comment by justinfx** [comment]
I think the idea of returning a new FileSequence is interesting. Although it seems inconsistent with a single index returning a string. If you wanted a sub-sequence you currently would have to call setFrameRange. Maybe a brand new method like "subsequence" or "slice" or "range"? That would accept range() syntax to return a new sequence range.

**Comment by justinfx** [comment]
I thought about this some more, and it felt wasteful in both time and memory to build a potentially huge list of strings for the case that the caller is just going to turn them into another sequence. The latest push in #61 returns a `FileSequence` when slicing. It will raise an `IndexError` if the slice is completely out of range. If the caller wants a list of files then they can easily call `list(fs)` on the returned sub-sequence, or just iterate on it.

**Comment by justinfx** [comment]
Merged #61

**Comment by assumptionsoup** [comment]
I agree, I like that decision, and I think it actually makes a lot of sense after thinking about it a bit more. I don't think it's inconsistent. When you index you expect a single item back, and when asking for a slice, you want an iterable, which in this case just happens to be a FileSequence.  I also like the new copy function.  That's something I was already looking for (I was subclassing and using deepcopy). Thanks Justin!
